iris/service/content/petition/mapper.py

```diff
@@ -51,6 +52,32 @@ def stateFilter(value):
     }
 
 
+def cityFilter(cityProperty):
+    """
+    city properties is either 'city.portal' or 'city.tags'
+    """
+    def wrapper(value):
+        cities = City.search({
+            "query": {
+                "filtered": {
+                    "filter": {
+                        "bool": {
+                            "must": queries.termsFilter(cityProperty)(value)
+                        }
+                    }
+                }
+            },
+            "fields": []
+        }, resolve_hits=False)['hits']['hits']
+        cityIds = [c['_id'] for c in cities]
+        return {
+            "bool": {
+                "should": {"terms": {"relations.city": cityIds}}
```

Comment:
Is this working?
The `should` parameter is a list of queries.
Anyway "bool" is not needed here, just return the terms filter.